src/main/java/com/example/petbuddybackend/service/chat/ChatService.java

```diff
@@ -138,6 +130,17 @@ private ChatMessageDTO createChatRoomWithMessageForCaretakerSender(
         return chatMapper.mapToChatMessageDTO(chatMessageRepository.save(chatMessage), timeZone);
     }
 
+    private ChatMessage createMessage(Long chatId, String principalEmail, Role principalRole, ChatMessageSent chatMessage) {
```

Comment:
nie podoba mi sie ze ta metoda nazywa sie createMessage, jest wywolywana przez createMessage i wywoluje createMessage

Reply:
Rekurencja :)

Zmieniłem te nazwy, faktycznie to źle wygląda